Sources/AWSXRayInstrument/Span.swift

```diff
@@ -0,0 +1,99 @@
+import AWSXRayRecorder
+import Baggage
+import Dispatch
+import Instrumentation
+
+extension XRayRecorder.Segment: Instrumentation.Span {
+    public var operationName: String { name }
+
+    public var kind: SpanKind {
+        // TODO: not sure if/how it corresponds to XRay segment
+        .internal
+    }
+
+    public var status: SpanStatus? {
+        // TODO: currently only setError() is public, expose Exception type or perhaps overload setError to provide message and type (as String)
+        // TODO: expose errors
+        get {
+            nil
+        }
+        set(newValue) {}
+    }
+
+    public var startTimestamp: DispatchTime {
+        // TODO: not sure if DispatchTime is the best choice here as it uses relative uptime
+        // XRayRecorder.Segment internally uses DispatchWallTime,
+        // currently it exposes seconds since which used to be internal
+        DispatchTime.now()
+    }
```

Comment:
@slashmo

Reply:
Whoops we messed up here, you're right it should be based on `DispatchWallTime`, fixing.
I think I'll hide it as a Timestamp type though, so people don't need to import dispatch _too_ to use the tracing APIs.

Reply:
Fixed in https://github.com/slashmo/gsoc-swift-tracing/pull/82 thanks again